src/test/java/com/github/tombentley/klog/segment/reader/SegmentDumpReaderTest.java

```diff
@@ -243,9 +243,9 @@ public void testWithTransactionLogDecoder() {
                     "baseOffset: 933607637 lastOffset: 933607638 count: 2 baseSequence: 0 lastSequence: 0 producerId: -1 producerEpoch: -1 partitionLeaderEpoch: 63 isTransactional: false isControl: false position: 0 CreateTime: 1655761268674 size: 165 magic: 2 compresscodec: NONE crc: 1118624748 isvalid: true\n" +
                     "| offset: 933607637 CreateTime: 1655761268674 keysize: 71 valuesize: 24 sequence: 0 headerKeys: []\n" +
                     "| offset: 933607638 CreateTime: 1655761268674 keysize: 71 valuesize: 24 sequence: 1 headerKeys: []\n",
-            // 3.2 dump
+            // 3.x dump
```

Comment:
It doesn't have to be in this PR, but if we're going to maintain this tool we might want to invest in having tests which cover dumps taken with a range of versions of `kafka-dump-logs.sh`.

Reply:
Agree. 

Given that there are still known EOS corner cases and possibly KRaft/2tiered corner cases in the future, this tool may still be useful for debugging.

I think it should be focused on issue and problematic pattern detection, which is what we did for EOS. For each item, it could also suggest possible fixes.